packages/core/src/main/notification/NotificationService.ts

```diff
@@ -52,29 +52,22 @@ export class NotificationService {
   }
 
   public async setLastEventDate(eventDate: Date): Promise<Date> {
+    let databaseLastEventDate: Date | undefined;
+
     try {
-      const databaseLastEventDate = await this.database.getLastEventDate();
-      if (eventDate > databaseLastEventDate) {
-        return this.database.updateLastEventDate(eventDate);
-      }
-      return databaseLastEventDate;
+      databaseLastEventDate = await this.database.getLastEventDate();
     } catch (error) {
-      if (
-        error instanceof StoreEngineError.RecordNotFoundError ||
-        error.constructor.name === StoreEngineError.RecordNotFoundError.constructor.name
-      ) {
-        return this.database.createLastEventDate(eventDate);
-      }
-      throw error;
+      return this.database.createLastEventDate(eventDate);
```

Comment:
Wouldn't that mean that we always try to create a lastEventDate no matter what kind of error is thrown?

Reply:
Indeed. I will add the check for the error type back.